fe/fe-core/src/main/java/org/apache/doris/scheduler/manager/TransientTaskManager.java

```diff
@@ -63,10 +63,8 @@ public Long addMemoryTask(TransientTaskExecutor executor) throws JobException {
     }
 
     public void cancelMemoryTask(Long taskId) throws JobException {
-        try {
+        if (taskExecutorMap.containsKey(taskId)) {
```

Comment:
I suggest to get the task from map first, and then check if it is null and do the cancel.
Because the task may be removed after `containsKey()` and before `get()`,
this 2 methods are not atomic